tests/e2e/pipeline.go

```diff
@@ -441,40 +461,42 @@ func deletePipelineRollout(name string) {
 }
 
 // update PipelineRollout and verify correct process
-func updatePipelineRollout(name string, newSpec numaflowv1.PipelineSpec, expectedFinalPhase numaflowv1.PipelinePhase, dataLoss bool, overrideSourceVertexReplicas bool) {
+func UpdatePipelineRollout(name string, newSpec numaflowv1.PipelineSpec, expectedFinalPhase numaflowv1.PipelinePhase, verifySpecFunc func(numaflowv1.PipelineSpec) bool, dataLoss bool, overrideSourceVertexReplicas bool) {
 
-	document("Updating Pipeline spec in PipelineRollout")
+	Document("Updating Pipeline spec in PipelineRollout")
 	rawSpec, err := json.Marshal(newSpec)
 	Expect(err).ShouldNot(HaveOccurred())
 
 	// update the PipelineRollout
-	updatePipelineRolloutInK8S(Namespace, name, func(rollout apiv1.PipelineRollout) (apiv1.PipelineRollout, error) {
+	UpdatePipelineRolloutInK8S(Namespace, name, func(rollout apiv1.PipelineRollout) (apiv1.PipelineRollout, error) {
 		rollout.Spec.Pipeline.Spec.Raw = rawSpec
 		return rollout, nil
 	})
 
 	// no data loss OR system pause
```

Comment:
why does it say "no data loss OR system pause"?

Reply:
`dataLoss` is a bit of a confusing param, it informs us if the update will cause data loss or not. If I remember correctly, I added `system pause` cause I was debugging this function when I was testing the manual pause case. In any case, this is a leftover comment we can remove for now.